ambry-server/src/main/java/com.github.ambry.server/AmbryServerRequests.java

```diff
@@ -0,0 +1,653 @@
+import com.github.ambry.protocol.BlobStoreControlAdminRequest;
+import com.github.ambry.protocol.CatchupStatusAdminRequest;
+import com.github.ambry.protocol.CatchupStatusAdminResponse;
+import com.github.ambry.protocol.ReplicationControlAdminRequest;
+import com.github.ambry.protocol.RequestControlAdminRequest;
+import com.github.ambry.protocol.RequestOrResponseType;
+import com.github.ambry.replication.FindTokenHelper;
+import com.github.ambry.replication.ReplicationAPI;
+import com.github.ambry.replication.ReplicationManager;
+import com.github.ambry.store.BlobStore;
+import com.github.ambry.store.StorageManager;
+import com.github.ambry.store.Store;
+import com.github.ambry.store.StoreException;
+import com.github.ambry.store.StoreKeyConverterFactory;
+import com.github.ambry.store.StoreKeyFactory;
+import com.github.ambry.utils.SystemTime;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The main request implementation class. All requests to the server are
+ * handled by this class
+ */
+public class AmbryServerRequests extends AmbryRequests {
+
+  private final Map<PartitionId, ReplicaId> localPartitionToReplicaMap;
+  private final StatsManager statsManager;
+  private final ConcurrentHashMap<RequestOrResponseType, Set<PartitionId>> requestsDisableInfo =
+      new ConcurrentHashMap<>();
+
+  private static final Logger logger = LoggerFactory.getLogger(AmbryServerRequests.class);
+
+  public AmbryServerRequests(StoreManager storeManager, RequestResponseChannel requestResponseChannel,
+      ClusterMap clusterMap, DataNodeId nodeId, MetricRegistry registry, ServerMetrics serverMetrics,
+      FindTokenHelper findTokenHelper, NotificationSystem operationNotification, ReplicationAPI replicationEngine,
+      StoreKeyFactory storeKeyFactory, boolean enableDataPrefetch, StoreKeyConverterFactory storeKeyConverterFactory,
+      StatsManager statsManager) {
+    super(storeManager, requestResponseChannel, clusterMap, nodeId, registry, serverMetrics, findTokenHelper,
+        operationNotification, replicationEngine, storeKeyFactory, enableDataPrefetch, storeKeyConverterFactory);
+    this.statsManager = statsManager;
+
+    for (RequestOrResponseType requestType : EnumSet.of(RequestOrResponseType.PutRequest,
+        RequestOrResponseType.GetRequest, RequestOrResponseType.DeleteRequest,
+        RequestOrResponseType.ReplicaMetadataRequest, RequestOrResponseType.TtlUpdateRequest)) {
+      requestsDisableInfo.put(requestType, Collections.newSetFromMap(new ConcurrentHashMap<>()));
+    }
+    localPartitionToReplicaMap = createLocalPartitionToReplicaMap();
```

Comment:
Just curious.. Why did we remove Collections.unmodifiableMap from here? Seemed like a good practice to have it.

Reply:
I made the change in my previous PR. The intention is to support dynamically adding new replica to server node which requires `localPartitionToReplicaMap` to be mutable. I will take a second look if I misunderstood semantic of `unmodifiableMap` here.

Reply:
Right I get it now. With your changes the number of local replicas will change, hence we it shouldn't be a unmodifiable map anymore. Makes sense!